internal/services/search/search_service_resource.go

```diff
@@ -209,21 +209,29 @@ func resourceSearchServiceCreateUpdate(d *pluginsdk.ResourceData, meta interface
 		properties.ServiceProperties.PartitionCount = utils.Int32(partitionCount)
 	}
 
-	future, err := client.CreateOrUpdate(ctx, resourceGroup, name, properties, nil)
+	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.Name, properties, nil)
 	if err != nil {
-		return fmt.Errorf("issuing create/update request for Search Service %q (ResourceGroup %q): %s", name, resourceGroup, err)
+		return fmt.Errorf("creating/updating %s: %v", id, err)
 	}
 
 	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
-		return fmt.Errorf("waiting for the completion of the creating/updating of Search Service %q (Resource Group %q): %+v", name, resourceGroup, err)
+		return fmt.Errorf("waiting for creation/update of %s: %+v", id, err)
 	}
 
-	resp, err := client.Get(ctx, resourceGroup, name, nil)
-	if err != nil {
-		return fmt.Errorf("issuing get request for Search Service %q (ResourceGroup %q): %s", name, resourceGroup, err)
+	timeout, _ := ctx.Deadline()
+
+	stateConf := &pluginsdk.StateChangeConf{
```

Comment:
hey @owenfarrell 

Just to double-check here, the Future is polling until Azure is marking this resource as `Succeeded` - as such this explicit check on `ProvisioningState` shouldn't be necessary (since the Future should be doing that) - are you seeing that this isn't provisioned in the `Succeeded` state when the resource is provisioned? If so, is that only in certain conditions?

Whilst we can look to apply this fix, we should look to [raise an upstream bug report about this](https://github.com/Azure/azure-rest-api-specs) so that this can be fixed in the upstream Azure API too.

Thanks!